ukfom/ukf.hpp

```diff
@@ -187,32 +187,21 @@ class ukf {
 	{
 		assert(X.size() == 2 * n + 1);
 
-		lapack::cholesky<n> L(sigma);
+        Eigen::LLT<cov> lltOfSigma(sigma); // compute the Cholesky decomposition of A
+        cov L = lltOfSigma.matrixL(); // retrieve factor L  in the decomposition
 
-		if (!L.isSPD())
-		{
-			std::cerr << std::endl << "sigma is not SPD:" << std::endl
-					  << sigma << std::endl
-					  << "---" << std::endl;
-			Eigen::EigenSolver<cov> eig(sigma);
-			std::cerr << "eigen values: " << eig.eigenvalues().transpose() << std::endl;
-		}
-		
-		assert(L.isSPD());
```

Comment:
This is what [guaranteeSPD](https://github.com/rock-core/base-types/blob/38662f835f555e5be6de69f80402ef76cbbb08b5/base/Eigen.hpp#L58
) method does. It set all negative Eigen values to 0.0